File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/exercicios/atividade_2/Atividade2.c
#include <stdio.h>

//Atuvudade que complementa o Exemplo1.c e Exemplo2.c
//Usar junto com o pub_tipo3.in

struct Duracao{
    int hora;
    int minuto;
};

typedef struct Duracao Duracao;

void ProcessarRegistro(int id, double preco, int tempo.hora, int tempo.minuto, char cidade[]){
    printf("Registro lido: %d %.2lf %d:%d %s\n", id, preco, tempo.hora. tempo.minuto, cidade);
}

int main(void){
    int id;
    double preco;
    Duracao tempo;
    char cidade[50];

    while(scanf(" %d", &id) == 1 && id != 0){
        scanf(" %lf %d %d %[^\n]", &preco, &tempo.hora, &tempo.minuto, cidade);
        ProcessarRegistro(id, preco, tempo.hora, tempo.minuto, cidade);
    }

    return 0;
}

File: Algoritimo-Estrutruras-de-Dados/Aeds-ll/2025_01/exercicios/atividade_2/Atividade2.java
import java.util.Locale;
import java.util.Scanner;

//Atuvudade que complementa o Exemplo1.java e Exemplo2.java
//Usar junto com o pub_tipo3.in

class Atividade2{
  public static void ProcessarRegistro(int id, double preco, String tempo, String cidade){
    System.out.printf("Registro lido: %d %.2f %s %s\n", id, preco, tempo, cidade);
  }

  public static void main(String[] args){
    Scanner scanner = new Scanner(System.in);
    int id;
    double preco;
    String tempo;
    String cidade;

    while((id = scanner.nextInt()) != 0){
      preco = scanner.nextDouble();
      tempo = scanner.next();
      cidade = scanner.nextLine().trim();
      ProcessarRegistro(id, preco, tempo, cidade);
    }

    scanner.close();
  }
}